src/java.base/windows/classes/sun/nio/ch/PipeImpl.java

```diff
@@ -162,28 +166,34 @@ public void run() {
                     try {
                         if (ssc != null)
                             ssc.close();
+                        if (sa instanceof UnixDomainSocketAddress uaddr) {
+                            Files.deleteIfExists(uaddr.getPath());
+                        }
                     } catch (IOException e2) {}
                 }
             }
         }
     }
 
     /**
-     * Creates a Pipe implementation that supports buffering.
+     * Creates a (TCP) Pipe implementation that supports buffering.
      */
     PipeImpl(SelectorProvider sp) throws IOException {
-        this(sp, true);
+        this(sp, true, false);
```

Comment:
I’m pretty sure that this should actually be:
```suggestion
        this(sp, /*AF_UNIX*/ false, /*buffering*/ true);
```

Reply:
Good catch. I will log a bug for that. In practice (and fortunately) it doesn't really matter since AFAICT this constructor is never called.

Reply:
> Good catch. I will log a bug for that. In practice (and fortunately) it doesn't really matter since AFAICT this constructor is never called.

The SelectorProvider's openPipe method creates the pipe implementation with the 1-arg constructor. The Selector creates the pipe for the wakeup mechanism with the 3-arg constructor. This PR is about the latter rather than the former. It is true that the comment in constructor description doesn't match the implementation but I read the comment from ExE-Boss to be asking for the openPipe to return an implementation that uses TCP sockets.